experimental/bakeable/src/lib.rs

```diff
@@ -0,0 +1,176 @@
+extern crate alloc;
+use alloc::borrow::Cow;
+use proc_macro2::TokenStream;
+use quote::quote;
+
+#[cfg(feature = "derive")]
+pub use bakeable_derive::Bakeable;
+
+pub trait Bakeable {
+    fn bake(&self) -> TokenStream;
+}
+
+macro_rules! literal {
+    ($($type:ty),*) => {
+        $(
+            impl Bakeable for $type {
+                fn bake(&self) -> TokenStream {
+                    quote! {
+                        #self
+                    }
+                }
+            }
+        )*
+    }
+}
+
+literal!(u8, u16, u32, u64, u128, usize, i8, i16, i32, i64, i128, isize, &str, str);
```

Comment:
question: should there be such an impl for `str`? You can't actually generate a `str` literal